core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -348,26 +348,52 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
       }
     for {
 
-      workflowsToAbort <- deletionFuture
+      workflowsToAbort <- deletionFuture recoverWith {
+        case t:Throwable => { // logging added as part of investigation into https://broadworkbench.atlassian.net/browse/CA-1206
+          logger.warn(s"Unexpected failure deleting workspace (while running `deletionFuture`) for workspace `${workspaceName}`", t)
+          throw t
+        }
+      }
 
       // Abort running workflows
-      aborts = Future.traverse(workflowsToAbort) { wf => executionServiceCluster.abort(wf, userInfo) }
+      aborts = Future.traverse(workflowsToAbort) { wf => executionServiceCluster.abort(wf, userInfo) } recoverWith {
+        case t:Throwable => { // logging added as part of investigation into https://broadworkbench.atlassian.net/browse/CA-1206
+          logger.warn(s"Unexpected failure deleting workspace (while aborting workflows) for workspace `${workspaceName}`", t)
+          throw t
+        }
+      }
 
       // Delete resource in sam outside of DB transaction
-      _ <- workspaceContext.workflowCollectionName.map( cn => samDAO.deleteResource(SamResourceTypeNames.workflowCollection, cn, userInfo) ).getOrElse(Future.successful(()))
+      _ <- workspaceContext.workflowCollectionName.map( cn => samDAO.deleteResource(SamResourceTypeNames.workflowCollection, cn, userInfo) ).getOrElse(Future.successful(())) recoverWith {
+        case t:Throwable => { // logging added as part of investigation into https://broadworkbench.atlassian.net/browse/CA-1206
+          logger.error(s"Unexpected failure deleting workspace (while deleting workflowCollection in Sam) for workspace `${workspaceName}`", t)
+          throw t
+        }
+      }
       // Delete workspace manager record (which will only exist if there had ever been a TDR snapshot in the WS)
-      _ = Try(workspaceManagerDAO.deleteWorkspace(workspaceContext.workspaceIdAsUUID, userInfo.accessToken)).recoverWith {
+      _ <- Future(workspaceManagerDAO.deleteWorkspace(workspaceContext.workspaceIdAsUUID, userInfo.accessToken)).recoverWith {
         //this will only ever succeed if a TDR snapshot had been created in the WS, so we gracefully handle all exceptions here
         case e: ApiException => {
           if(e.getCode != StatusCodes.NotFound.intValue) {
-            logger.warn(s"Unexpected failure deleting workspace in Workspace Manager. Received ${e.getCode}: [${e.getResponseBody}]")
+            logger.warn(s"Unexpected failure deleting workspace (while deleting in Workspace Manager) for workspace `${workspaceName}. Received ${e.getCode}: [${e.getResponseBody}]")
+            Future.successful()
           }
-          Success(())
+          else throw e
```

Comment:
Looking for a review from AS on this

Reply:
* is there a reason to move this from a `Try` to a `Future`? Either works, and if you want to use `Future` for consistency, I understand … is there some other reason?
* the logic as you have written it is incorrect; you are swallowing all non-404 errors and rethrowing all 404s. This is backwards - we should swallow 404s. I think this is why you have so many test failures.
* you're correct that you're making this part of the code stricter. As originally written, we swallowed ALL errors but only logged the non-404s; now we will throw on non-404s (after fixing the previous bullet point). This feels like a good change, now that workspace manager has been live and stable for a while.